create.go

```diff
@@ -177,11 +208,11 @@ func (r *Resource) computeCreateEventPatches(ctx context.Context, accessor metav
 	// an update is about to be processed. So we set the status condition
 	// indicating the guest cluster is updating now.
 	{
-		isNotEmpty := currentVersion != ""
-		isNotUpdating := !clusterStatus.HasUpdatingCondition()
+		isCreated := clusterStatus.HasCreatedCondition()
+		notUpdating := !clusterStatus.HasUpdatingCondition()
 		versionDiffers := currentVersion != desiredVersion
 
-		if isNotEmpty && isNotUpdating && versionDiffers {
+		if isCreated && notUpdating && versionDiffers {
```

Comment:
The version and updating status management was combined and is now separated. Should be more reliable and independent from each other.